SU2_CFD/include/solvers/CFVMFlowSolverBase.hpp

```diff
@@ -244,6 +247,257 @@ class CFVMFlowSolverBase : public CSolver {
    */
   inline virtual void InstantiateEdgeNumerics(const CSolver* const* solvers, const CConfig* config) {}
 
+  /*!
+   * \brief Generic implementation to compute the time step based on CFL and conv/visc eigenvalues.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] solver_container - Container vector with all the solutions.
+   * \param[in] config - Definition of the particular problem.
+   * \param[in] iMesh - Index of the mesh in multigrid computations.
+   * \param[in] Iteration - Value of the current iteration.
+   * \tparam SoundSpeedFunc - Function object to compute speed of sound
+   * \tparam LambdaViscFunc - Function object to compute the viscous lambda
+   * \note Both functors need to implement (nodes,iPoint,jPoint) for edges, and (nodes,iPoint) for vertices.
+   */
+  template<class SoundSpeedFunc, class LambdaViscFunc>
+  FORCEINLINE void SetTime_Step_impl(const SoundSpeedFunc& soundSpeed,
+                                     const LambdaViscFunc& lambdaVisc,
+                                     CGeometry *geometry,
```

Comment:
The idea is to generalize this type of function, the details of the implementation are provided by the derived solver when it instantiates (calls) the generic implementation.
There is no explicit "if incompressible" or "if nemo" or "if compressible" logic.

Reply:
Thanks for going through this...I made some attempts to update time integration schemes in another branch but didnt push yet.

Reply:
oops sorry about that, I was looking at this kind of routine and had a "what if" moment... and now it's done

Reply:
No apologies! Im happy its done.